common/database_handler.py

```diff
@@ -50,10 +50,20 @@ def close_all_connections(cls):
         """Close all connections."""
         cls.pool.closeall()
 
+    @classmethod
+    def recover_connection(cls, size):
+        """Recover connection to database"""
+        cls.close_all_connections()
+        cls.create_connection_pool(size)
+
 
 class DatabasePool:
     """Context manager for connection pool."""
+
+    POOL_SIZE = 0
```

Comment:
maybe it's better to make it `DatabasePoolHandler` class attribute, together on same level as `pool` attribute and set the size in `create_connection_pool` function directly